text_collector_examples/nvme_metrics.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+set -eu
+
+# Dependencies: nvme-cli, jq (packages)
+# Based on code from
+# - https://github.com/prometheus/node_exporter/blob/master/text_collector_examples/smartmon.sh
+# - https://github.com/vorlon/check_nvme/blob/master/check_nvme.sh
+# Author: Henk <henk@wearespindle.com>
+
+output_format_awk="$(
+  cat <<'OUTPUTAWK'
+BEGIN { v = "" }
+v != $1 {
+  print "# HELP nvme_" $1 " SMART metric " $1;
+  print "# TYPE nvme_" $1 " gauge";
+  v = $1
+}
+{print "nvme_" $0}
+OUTPUTAWK
+)"
+
+format_output() {
+  sort | awk -F'{' "${output_format_awk}"
+}
+
+# Get the nvme-cli version
+nvme_version="$(nvme version | awk '$1 == "nvme" {print $3}')"
+echo "nvmecli{version=\"${nvme_version}\"} 1" | format_output
+
+# Cut device name to /dev/nvmeX
+device_list="$(nvme list | awk '/^\/dev/{print $1}' | cut -c1-10)"
+
+# Loop through the NVMe devices
+for device in ${device_list}; do
+  normal_check="$(nvme smart-log "${device}")"
+  json_check="$(nvme smart-log -o json "${device}")"
+  disk="$(echo "${device}" | cut -c6-10)"
+
+  # Normal output checks ######################################################
+  # Converting percentage to ratio is easier with awk
+  value_available_spare="$(echo "$normal_check" | sed 's/%$//' | awk '$1 == "available_spare" {print $3 / 100}')"
+  echo "available_spare_ratio{disk=\"${disk}\"} ${value_available_spare}"
+
+  value_available_spare_threshold="$(echo "$normal_check" | sed 's/%$//' | awk '$1 == "available_spare_threshold" {print $3 / 100}')"
+  echo "available_spare_threshold_ratio{disk=\"${disk}\"} ${value_available_spare_threshold}"
+
+  value_percentage_used="$(echo "$normal_check" | sed 's/%$//' | awk '$1 == "percentage_used" {print $3 / 100}')"
+  echo "percentage_used_ratio{disk=\"${disk}\"} ${value_percentage_used}"
+
+  # The temperature value in JSON is odd, hence checking with "normal"
+  value_temperature="$(echo "$normal_check" | awk '$1 == "temperature" {print $3}')"
```

Comment:
`jq '.temperature - 273'` should solve the Kelvin conversion.

Reply:
I was too quick with pushing and then read this comment... Did something similar but more "complicated". Will fix that.

Thanks for you patience - this has been somewhat of a challenge for me :)

Reply:
No problem, thanks for the contribution.